platform/core/src/capabilities/PersistenceCapability.js

```diff
@@ -148,13 +148,19 @@ define(
          */
         PersistenceCapability.prototype.refresh = function () {
             var domainObject = this.domainObject;
+            var $q = this.$q;
 
             // Update a domain object's model upon refresh
             function updateModel(model) {
-                var modified = model.modified;
-                return domainObject.useCapability("mutation", function () {
-                    return model;
-                }, modified);
+                if (model === undefined) {
```

Comment:
This case isn't handled by previous code; and as I see it, this case shouldn't ever occur.  Is there a reason for this check?